src/controllers/dlgprefcontroller.cpp

```diff
@@ -992,25 +988,51 @@ void DlgPrefController::slotShowPreviewScreens(
 #endif
 
 void DlgPrefController::slotShowMapping(std::shared_ptr<LegacyControllerMapping> pMapping) {
-    m_ui.labelLoadedMapping->setText(mappingName(pMapping));
-    m_ui.labelLoadedMappingDescription->setText(mappingDescription(pMapping));
-    m_ui.labelLoadedMappingAuthor->setText(mappingAuthor(pMapping));
-    m_ui.labelLoadedMappingSupportLinks->setText(mappingSupportLinks(pMapping));
-    m_ui.labelLoadedMappingScriptFileLinks->setText(mappingFileLinks(pMapping));
+    QString name, description, author, supportLinks, scriptFileLinks;
+
+    if (pMapping) {
+        name = pMapping->name();
+        description = pMapping->description();
+        author = pMapping->author();
+        supportLinks = mappingSupportLinks(pMapping);
+        scriptFileLinks = mappingFileLinks(pMapping);
+    }
+
+    m_ui.labelMappingNameValue->setText(name);
+    m_ui.labelMappingDescriptionValue->setText(description);
+    m_ui.labelMappingAuthorValue->setText(author);
+    m_ui.labelMappingSupportLinks->setText(supportLinks);
+    m_ui.labelMappingScriptFileLinks->setText(scriptFileLinks);
+
+    // Hide or show labels based on the presence of text
+    m_ui.labelMappingName->setVisible(!name.isEmpty());
+    m_ui.labelMappingNameValue->setVisible(!name.isEmpty());
+    m_ui.labelMappingDescription->setVisible(!description.isEmpty());
+    m_ui.labelMappingDescriptionValue->setVisible(!description.isEmpty());
+    m_ui.labelMappingAuthor->setVisible(!author.isEmpty());
+    m_ui.labelMappingAuthorValue->setVisible(!author.isEmpty());
+    m_ui.labelMappingSupport->setVisible(!supportLinks.isEmpty());
+    m_ui.labelMappingScriptFiles->setVisible(!scriptFileLinks.isEmpty());
+    // We do not hide the support links label and
+    // the script files label if they are empty,
+    // because without them the layout would be collapse.
+
+    // Remove all tabs first
+    m_ui.controllerTabs->clear();
```

Comment:
Why remove and insert tabs?
There is also [`QTabWidget::setTabVisible(int index, bool visible)`](https://doc.qt.io/qt-6/qtabwidget.html#setTabVisible)
IIUC tab indices are unchanged (unlike visual position in ui) so we'd only need to store them in the ctor.

This way we'd also get around double tab tr strings (here and in ui file)

Reply:
same applies to the screens and settings tab, of course.